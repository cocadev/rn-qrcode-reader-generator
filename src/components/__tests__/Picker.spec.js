/* eslint-disable no-undef */
import React from 'react';
import { shallow } from 'enzyme';

import {
  Picker,
} from '../index';

describe('Picker Component', () => {
  it('renders as expected on datetime', () => {
    const props = {
      type: 'datetime',
      key: 1,
      placeholder: 'hello',
    };

    const wrapper = shallow(
      <Picker {...props} />,
    );
    expect(wrapper).toMatchSnapshot();
  });

  it('renders as expected on picker', () => {
    const props = {
      type: 'picker',
      key: 1,
      items: [
        { id: 1, label: 'hello' },
        { id: 2, label: 'man' },
      ],
    };

    const wrapper = shallow(
      <Picker {...props} />,
    );
    expect(wrapper).toMatchSnapshot();
  });
});
