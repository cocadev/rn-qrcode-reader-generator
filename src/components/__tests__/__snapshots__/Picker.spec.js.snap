// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Picker Component renders as expected on datetime 1`] = `
ShallowWrapper {
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <withState(withState(withHandlers(withHandlers(PickerComponent))))
    placeholder="hello"
    type="datetime"
  />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateError": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "class",
    "props": Object {
      "inputValue": "",
      "placeholder": "hello",
      "setInputValue": [Function],
      "type": "datetime",
    },
    "ref": null,
    "rendered": null,
    "type": [Function],
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "class",
      "props": Object {
        "inputValue": "",
        "placeholder": "hello",
        "setInputValue": [Function],
        "type": "datetime",
      },
      "ref": null,
      "rendered": null,
      "type": [Function],
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
        "lifecycles": Object {
          "componentDidUpdate": Object {
            "onSetState": true,
          },
          "getDerivedStateFromProps": true,
          "getSnapshotBeforeUpdate": true,
          "setState": Object {
            "skipsComponentDidUpdateOnNullish": true,
          },
        },
      },
    },
  },
}
`;

exports[`Picker Component renders as expected on picker 1`] = `
ShallowWrapper {
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <withState(withState(withHandlers(withHandlers(PickerComponent))))
    items={
      Array [
        Object {
          "id": 1,
          "label": "hello",
        },
        Object {
          "id": 2,
          "label": "man",
        },
      ]
    }
    type="picker"
  />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateError": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "class",
    "props": Object {
      "inputValue": "",
      "items": Array [
        Object {
          "id": 1,
          "label": "hello",
        },
        Object {
          "id": 2,
          "label": "man",
        },
      ],
      "setInputValue": [Function],
      "type": "picker",
    },
    "ref": null,
    "rendered": null,
    "type": [Function],
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "class",
      "props": Object {
        "inputValue": "",
        "items": Array [
          Object {
            "id": 1,
            "label": "hello",
          },
          Object {
            "id": 2,
            "label": "man",
          },
        ],
        "setInputValue": [Function],
        "type": "picker",
      },
      "ref": null,
      "rendered": null,
      "type": [Function],
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
        "lifecycles": Object {
          "componentDidUpdate": Object {
            "onSetState": true,
          },
          "getDerivedStateFromProps": true,
          "getSnapshotBeforeUpdate": true,
          "setState": Object {
            "skipsComponentDidUpdateOnNullish": true,
          },
        },
      },
    },
  },
}
`;
